superset/assets/src/explore/components/ExploreChartPanel.jsx

```diff
@@ -40,38 +39,47 @@ class ExploreChartPanel extends React.PureComponent {
   renderChart() {
     const { chart } = this.props;
     return (
-      <ChartContainer
-        chartId={chart.id}
-        containerId={this.props.containerId}
-        datasource={this.props.datasource}
-        formData={this.props.form_data}
-        height={this.getHeight()}
-        slice={this.props.slice}
-        setControlValue={this.props.actions.setControlValue}
-        timeout={this.props.timeout}
-        vizType={this.props.vizType}
-        refreshOverlayVisible={this.props.refreshOverlayVisible}
-        errorMessage={this.props.errorMessage}
-        onQuery={this.props.onQuery}
-        onDismissRefreshOverlay={this.props.onDismissRefreshOverlay}
-        annotationData={chart.annotationData}
-        chartAlert={chart.chartAlert}
-        chartStatus={chart.chartStatus}
-        chartUpdateEndTime={chart.chartUpdateEndTime}
-        chartUpdateStartTime={chart.chartUpdateStartTime}
-        latestQueryFormData={chart.latestQueryFormData}
-        lastRendered={chart.lastRendered}
-        queryResponse={chart.queryResponse}
-        queryController={chart.queryController}
-        triggerQuery={chart.triggerQuery}
-      />
+      <ParentSize>
+        {({ width, height }) => (width > 0 && height > 0) && (
+          <ChartContainer
+          chartId={chart.id}
+          containerId={this.props.containerId}
+          datasource={this.props.datasource}
+          formData={this.props.form_data}
+          width={Math.floor(width)}
+          height={this.getHeight()}
```

Comment:
should this now take the `height` from `ParentSize` into account?

Reply:
I just tried. It doesn't work out of the box. May need to change the css of the container (`.panel-header` and `.panel-body`) into `flexbox` similar to what I did with `WithLegend` to make sure `ParentSize` computes the correct `height`. Now the height is larger than it should be. 

I prefer not to deal with it now and leave this for future PR.

Reply:
sgtm 👍